### PR TITLE
Handle newly_created gracefully

### PR DESCRIPTION
**Pull request by MinnDevelopment**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [x] Internal code
- [ ] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

Adds handling for `newly_created` on THREAD_CREATE.

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/internal/handle/ThreadCreateHandler.java [hunk 1]
```diff
@@ -35,9 +36,14 @@ protected Long handleInternally(DataObject content)
         if (api.getGuildSetupController().isLocked(guildId))
             return guildId;
 
+        if (api.getThreadChannelById(content.getUnsignedLong("id")) != null)
+            return null;
```
Comment (DV8FromTheWorld) [inline src/main/java/net/dv8tion/jda/internal/handle/ThreadCreateHandler.java:40]: When would this happen?

Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/internal/handle/ThreadCreateHandler.java:40]: Without wanting to test every edge case, I'm assuming the event also fires in some cases when the member is added despite already knowing it exists (moderator permissions). I just want to account for any unexpected situations this way.

Comment (DV8FromTheWorld) [reply src/main/java/net/dv8tion/jda/internal/handle/ThreadCreateHandler.java]: Lets drop a comment on there indicating that this is guess-based protection instead of specifically known situation based protection
